File: src/test/java/com/raychenon/leetcode/IntersectionOfSortedArraysTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 2019-10-06
 * Time: 14:39
 */
public class IntersectionOfSortedArraysTest {


    private static IntersectionOfSortedArrays instance;

    @BeforeClass
    public static void setUp() {
        instance = new IntersectionOfSortedArrays();
    }

    @Test
    public void threeSortedArraysTest() {
        int arr1[] = {1, 3, 5, 6};
        int arr2[] = {1, 2, 5, 7, 9};
        int arr3[] = {1, 3, 4, 5, 8};

        List<Integer> expectedResults = Arrays.asList(1, 5);

        List<Integer> results = instance.arraysIntersection(arr1, arr2, arr3);
        assertEquals(expectedResults, results);
    }


    @Test
    public void noIntersectionInArraysTest() {
        int arr1[] = {1, 3, 5, 6};
        int arr2[] = {2, 7};
        int arr3[] = {8};

        List<Integer> expectedResults = Arrays.asList();

        List<Integer> results = instance.arraysIntersection(arr1, arr2, arr3);
        assertEquals(expectedResults, results);
    }

}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/IntersectionOfSortedArraysTest.kt
package com.raychenon.kotlin.leetcode

import junit.framework.TestCase.assertEquals
import org.junit.Test

/**
 * User: raychenon
 * Date: 2019-10-06
 * https://leetcode.com/problems/intersection-of-three-sorted-arrays/
 */
class IntersectionOfSortedArraysTest {

    @Test
    fun threeSortedArraysTest() {
        val arr1 = intArrayOf(1, 3, 5, 6)
        val arr2 = intArrayOf(1, 2, 5, 7, 9)
        val arr3 = intArrayOf(1, 3, 4, 5, 8)

        val expectedResults = listOf(1, 5)

        val results = IntersectionOfSortedArrays.arraysIntersection(arr1, arr2, arr3)
        assertEquals(expectedResults, results)
    }


    @Test
    fun noIntersectionInArraysTest() {
        val arr1 = intArrayOf(1, 3, 5, 6)
        val arr2 = intArrayOf(2, 7)
        val arr3 = intArrayOf(8)

        val expectedResults = emptyList<Int>()

        val results = IntersectionOfSortedArrays.arraysIntersection(arr1, arr2, arr3)
        assertEquals(expectedResults, results)
    }

}